bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/aWATTarBridgeHandler.java

```diff
@@ -0,0 +1,283 @@
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.openhab.binding.awattar.internal.aWATTarBridgeConfiguration;
+import org.openhab.binding.awattar.internal.aWATTarPrice;
+import org.openhab.binding.awattar.internal.connection.aWATTarConnectionException;
+import org.openhab.binding.awattar.internal.dto.AwattarApiData;
+import org.openhab.binding.awattar.internal.dto.Datum;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link aWATTarBridgeHandler} is responsible for retrieving data from the aWATTar API.
+ *
+ * The API provides hourly prices for the current day and, starting from 14:00, hourly prices for the next day.
+ * Check the documentation at https://www.awattar.de/services/api
+ *
+ *
+ *
+ * @author Wolfgang Klimt - Initial contribution
+ */
+@NonNullByDefault
+public class aWATTarBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(aWATTarBridgeHandler.class);
+    private final HttpClient httpClient;
+    @Nullable
+    private ScheduledFuture<?> dataRefresher;
+
+    private final String URL_DE = "https://api.awattar.de/v1/marketdata";
+    private final String URL_AT = "https://api.awattar.at/v1/marketdata";
+    private String URL;
+
+    // This cache stores price data for up to two days
+    @Nullable
+    private SortedMap<Long, aWATTarPrice> priceMap;
+    private final int dataRefreshInterval = 60;
+    private double vatFactor = 0;
+    private long lastUpdated = 0;
+    private double basePrice = 0;
+    private long minTimestamp = 0;
+    private long maxTimestamp = 0;
+    private ZoneId zone;
+
+    public aWATTarBridgeHandler(Bridge thing, HttpClient httpClient) {
+        super(thing);
+        logger.trace("Creating aWATTarBridgeHandler instance {}", this);
+        this.httpClient = httpClient;
+        URL = URL_DE;
+        zone = ZoneId.systemDefault();
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing aWATTar bridge {}", this);
+        updateStatus(ThingStatus.UNKNOWN);
+        aWATTarBridgeConfiguration config = getConfigAs(aWATTarBridgeConfiguration.class);
+        vatFactor = 1 + (config.vatPercent / 100);
+        basePrice = config.basePrice;
+        try {
+            zone = ZoneId.of(config.timeZone);
+            logger.trace("Using time zone {}", zone);
+        } catch (ZoneRulesException ex) {
+            logger.error("Zone ID not found: {}, {}", zone, ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/error.timezone");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Please check all.

Reply:
Checked and removed. I hope the few that remain are ok.